plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -863,25 +872,47 @@ public static Transaction newCreateTableTransaction(TrinoCatalog catalog, Connec
         PartitionSpec partitionSpec = parsePartitionFields(schema, getPartitioning(tableMetadata.getProperties()));
         SortOrder sortOrder = parseSortFields(schema, getSortOrder(tableMetadata.getProperties()));
 
+        Transaction txn;
+
         if (replace) {
-            return catalog.newCreateOrReplaceTableTransaction(session, schemaTableName, schema, partitionSpec, sortOrder, tableLocation, createTableProperties(session, tableMetadata, allowedExtraProperties));
+            txn = catalog.newCreateOrReplaceTableTransaction(session, schemaTableName, schema, partitionSpec, sortOrder, tableLocation, createTableProperties(tableMetadata, allowedExtraProperties));
         }
-        return catalog.newCreateTableTransaction(session, schemaTableName, schema, partitionSpec, sortOrder, Optional.ofNullable(tableLocation), createTableProperties(session, tableMetadata, allowedExtraProperties));
+        else {
+            txn = catalog.newCreateTableTransaction(session, schemaTableName, schema, partitionSpec, sortOrder, Optional.ofNullable(tableLocation), createTableProperties(tableMetadata, allowedExtraProperties));
+        }
+
+        // If user doesn't set compression-codec for parquet, we need to remove write.parquet.compression-codec property,
+        // Otherwise Iceberg will set write.parquet.compression-codec to zstd by default.
+        String parquetCompressionValue = txn.table().properties().get(PARQUET_COMPRESSION);
+        if (parquetCompressionValue != null && parquetCompressionValue.isEmpty()) {
+            txn.updateProperties()
+                    .remove(PARQUET_COMPRESSION)
+                    .commit();
```

Comment:
Calling `.commit()` here seems wrong since we don't want to commit the returned transaction until after the create table operation is ready to commit- or am I misunderstanding the semantics of the Iceberg library on this part?

Reply:
Maybe I'm misunderstanding the implication of calling `.commit()` since I see there also exists a `Transaction#commitTransaction()` method.